utils/interfaces/_core.py

```diff
@@ -88,6 +89,13 @@ def load_data_from_logs(self, folder_path):
                 with open(file_path, "r", encoding="utf-8") as f:
                     data = json.load(f)
 
+                # Apply the same deserialization process as in ingestion, since
+                # there may be a race condition where additional messages are written
+                # post-setup without prior normalization.
+                # XXX: A more robust solution would be moving the serialization process
+                # to the proxy, to avoid the need to rewrite request files in the first place.
+                deserialize(data, self.name)
```

Comment:
At this point, data may be deserialized. You deserialize it twice no ?

Reply:
As far as I can tell, `deserialize` is idempotent. So there's no harm in doing it twice. On the other hand, even if it was deserialized before, the `data` as read from the object vs the `data` after calling `deserialize` is not the same. For example, byte arrays will be strings like this `b'mybytestrict'` in the JSON, but after calling `deserialize`, they will be converted to actual byte arrays.